pkg/apis/extensions/types.go

```diff
@@ -437,6 +449,22 @@ type RollingUpdateDaemonSet struct {
 	MaxUnavailable intstr.IntOrString
 }
 
+// Spec to control the desired behavior of a daemon set surging rolling update.
+type SurgingRollingUpdateDaemonSet struct {
```

Comment:
That was proposed, but shot down at some stage
cc @janetkuo @erictune @roberthbailey 

see https://github.com/kubernetes/kubernetes/issues/48841 as well